url/src/lib.rs

```diff
@@ -349,11 +351,144 @@ impl Url {
     /// with this URL as the base URL, a [`ParseError`] variant will be returned.
     ///
     /// [`ParseError`]: enum.ParseError.html
+    /// [`make_relative`]: #method.make_relative
     #[inline]
     pub fn join(&self, input: &str) -> Result<Url, crate::ParseError> {
         Url::options().base_url(Some(self)).parse(input)
     }
 
+    /// Creates a relative URL if possible, with this URL as the base URL.
+    ///
+    /// This is the inverse of [`join`].
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use url::Url;
+    /// # use url::ParseError;
+    ///
+    /// # fn run() -> Result<(), ParseError> {
+    /// let base = Url::parse("https://example.net/a/b.html")?;
+    /// let url = Url::parse("https://example.net/a/c.png")?;
+    /// let relative = base.make_relative(&url);
+    /// assert_eq!(relative.as_deref(), Some("c.png"));
```

Comment:
```
---- src/lib.rs - Url::make_relative (line 366) stdout ----
error[E0599]: no method named `as_deref` found for type `std::option::Option<std::string::String>` in the current scope
```

What's the minimum supported Rust version here? `Option::as_deref()` is stable since 1.40

Reply:
CI runs with 1.36.0.

Reply:
Do you want to keep that, and if so, which of the many patterns to solve this would you prefer? `relative.as_ref().map(|s| s.as_str())`? :)

Reply:
This one seems fine.

Reply:
Done, thanks!